### PR TITLE
fix:fit `check_factorial.cpp` to guidelines

### PR DESCRIPTION
**Pull request by ewd00010**
#### Description of Change
FIX: added namespace math, changed for a weird for loop to a while loop with a better conditional and I initiation
CHORE: Cleaned documentation and added details plus links. removed the invalid cout lines in test for one line.

#### Checklist
- [x] Added description of change
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: clang-tidy check needed

### CODE REVIEW

File: math/check_factorial.cpp [hunk 1]
```diff
@@ -1,64 +1,72 @@
...
+ *
+ * @details A factorial number is the sum of k! where any value of k is a
+ * positive integer. https://www.mathsisfun.com/numbers/factorial.html
+ *
  * @author [Divyajyoti Ukirde](https://github.com/divyajyotiuk)
  */
-#include <cassert>
-#include <iostream>
+#include <cassert>   // for assert
+#include <iostream>  // for cout
```
Comment (realstealthninja) [inline math/check_factorial.cpp:12]:
```suggestion
#include <cassert>   /// for assert
#include <iostream>  /// for cout
```

File: math/check_factorial.cpp [hunk 2]
```diff
@@ -1,64 +1,72 @@
 /**
  * @file
  * @brief A simple program to check if the given number is a factorial of some
- * number or not.
+ * number or not. https://en.wikipedia.org/wiki/Factorial
```
Comment (realstealthninja) [inline math/check_factorial.cpp:4]:
```suggestion
 * number or not.
```

File: math/check_factorial.cpp [hunk 3]
```diff
@@ -1,64 +1,72 @@
 /**
  * @file
  * @brief A simple program to check if the given number is a factorial of some
```
Comment (realstealthninja) [inline math/check_factorial.cpp:3]:
```suggestion
 * @brief A simple program to check if the given number is a [factorial](https://en.wikipedia.org/wiki/Factorial) of some
```
hyperlinks seem more neat in this case

**Review by realstealthninja — CHANGES_REQUESTED** [review]
Thank you so much for helping us keep the repository up to par with the standards :heart:

**Comment by realstealthninja** [comment]
part of #2456

**Review by realstealthninja — APPROVED** [review]
LGTM!

File: math/check_factorial.cpp [hunk 4]
```diff
@@ -1,64 +1,72 @@
...
+ * positive integer. https://www.mathsisfun.com/numbers/factorial.html
+ *
  * @author [Divyajyoti Ukirde](https://github.com/divyajyotiuk)
  */
-#include <cassert>
-#include <iostream>
+#include <cassert>   /// for assert
+#include <iostream>  /// for cout
 
+namespace math {
```
Comment (Panquesito7) [inline math/check_factorial.cpp:14]: Missing namespace Doxygen documentation.
```suggestion
/**
 * @namespace
 * @brief Mathematical algorithms
 */
namespace math {
```

File: math/check_factorial.cpp [hunk 5]
```diff
@@ -1,64 +1,72 @@
...
-    std::cout << "passed\n";
-
-    std::cout << "Test 4:\t n=479001600\n";
-    assert(is_factorial(479001600) == true);
-    std::cout << "passed\n";
-
-    std::cout << "Test 5:\t n=-24\n";
-    assert(is_factorial(-24) == false);
-    std::cout << "passed\n";
+    std::cout << "all tests have passed successfully" << std::endl;
```
Comment (Panquesito7) [inline math/check_factorial.cpp:62]:
```suggestion
    std::cout << "All tests have successfully passed!" << std::endl;
```

File: math/check_factorial.cpp [hunk 6]
```diff
@@ -1,64 +1,72 @@
...
     }
 }
+}  // namespace math
 
-/** Test function
+/**
+ * @brief Test function
  * @returns void
  */
 void tests() {
```
Comment (Panquesito7) [inline math/check_factorial.cpp:54]:
```suggestion
static void tests() {
```

File: math/check_factorial.cpp [hunk 7]
```diff
@@ -1,64 +1,72 @@
...
         return true;
     } else {
         return false;
     }
 }
+}  // namespace math
 
-/** Test function
+/**
+ * @brief Test function
```
Comment (Panquesito7) [inline math/check_factorial.cpp:51]:
```suggestion
 * @brief Self-test implementations
```

File: math/check_factorial.cpp [hunk 8]
```diff
@@ -1,64 +1,72 @@
...
 }
 
-/** Main function
+/**
+ * @brief Main function
  * @returns 0 on exit
  */
 int main() {
-    tests();
+    tests();  // runs self-test implementation
```
Comment (Panquesito7) [inline math/check_factorial.cpp:70]:
```suggestion
    tests();  // run self-test implementations
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]

File: math/check_factorial.cpp [hunk 9]
```diff
@@ -1,64 +1,76 @@
 /**
  * @file
- * @brief A simple program to check if the given number is a factorial of some
+ * @brief A simple program to check if the given number is a [factorial](https://en.wikipedia.org/wiki/Factorial) of some
  * number or not.
+ *
+ * @details A factorial number is the sum of k! where any value of k is a
+ * positive integer. https://www.mathsisfun.com/numbers/factorial.html
+ *
  * @author [Divyajyoti Ukirde](https://github.com/divyajyotiuk)
```
Comment (Panquesito7) [inline math/check_factorial.cpp:9]:
```suggestion
 * @author [Divyajyoti Ukirde](https://github.com/divyajyotiuk)
 * @author [ewd00010](https://github.com/ewd00010)
```

**Review by Panquesito7 — APPROVED** [review]
Awesome. Thanks! 🚀

File: math/check_factorial.cpp [hunk 10]
```diff
@@ -1,64 +1,77 @@
...
+    while (n % i == 0) {
         n = n / i;
+        i++;
     }
+
+    /*!
+     * if n was the sum of a factorial then it should be divided until it
+     * becomes 1
+     */
     if (n == 1) {
```
Comment (alexpantyukhin) [inline math/check_factorial.cpp:47]: maybe
```cpp
    return (n == 1);
```
?

**Review by alexpantyukhin — APPROVED** [review]

**Review by Panquesito7 — DISMISSED** [review]
Awesome. Thanks! 🚀

File: math/check_factorial.cpp [hunk 11]
```diff
@@ -1,64 +1,73 @@
...
-        return true;
-    } else {
-        return false;
-    }
+
+    /*!
+     * if n was the sum of a factorial then it should be divided until it
+     * becomes 1
+     */
+    return (n == 1)
```
Comment (Panquesito7) [inline math/check_factorial.cpp:47]: There seems to be an error here. I think there's a missing `;`.
```suggestion
    return (n == 1);
```

Comment (ewd00010) [reply math/check_factorial.cpp:47]: the classic

**Review by Panquesito7 — APPROVED** [review]
Great work. Thanks! 🚀

**Review by realstealthninja — APPROVED** [review]
Great work!

**Pull request merged**